packages/kbn-securitysolution-grouping/src/hooks/use_get_group_selector.tsx

```diff
@@ -112,33 +116,46 @@ export const useGetGroupSelector = ({
 
   const onChange = useCallback(
     (groupSelection: string) => {
+      let newSelectedGroups: string[] = [];
+      let sendTelemetry = true;
       // Simulate a toggle behavior when maxGroupingLevels is 1
       if (maxGroupingLevels === 1) {
-        setSelectedGroups([groupSelection]);
+        newSelectedGroups = [groupSelection];
       } else {
+        // if the group is already selected, remove it
         if (selectedGroups.find((selected) => selected === groupSelection)) {
+          sendTelemetry = false;
           const groups = selectedGroups.filter((selectedGroup) => selectedGroup !== groupSelection);
           if (groups.length === 0) {
-            setSelectedGroups(['none']);
+            newSelectedGroups = ['none'];
           } else {
-            setSelectedGroups(groups);
+            newSelectedGroups = groups;
           }
-          return;
+        } else {
+          newSelectedGroups = isNoneGroup([groupSelection])
+            ? [groupSelection]
+            : [
+                ...selectedGroups.filter((selectedGroup) => selectedGroup !== 'none'),
+                groupSelection,
+              ];
         }
-
-        const newSelectedGroups = isNoneGroup([groupSelection])
-          ? [groupSelection]
-          : [...selectedGroups.filter((selectedGroup) => selectedGroup !== 'none'), groupSelection];
-        setSelectedGroups(newSelectedGroups);
       }
 
-      // built-in telemetry: UI-counter
-      tracker?.(
-        METRIC_TYPE.CLICK,
-        getTelemetryEvent.groupChanged({ groupingId, selected: groupSelection })
-      );
+      setSelectedGroups(newSelectedGroups);
 
-      onGroupChange?.({ tableId: groupingId, groupByField: groupSelection });
+      if (sendTelemetry) {
+        // built-in telemetry: UI-counter
+        tracker?.(
+          METRIC_TYPE.CLICK,
+          getTelemetryEvent.groupChanged({ groupingId, selected: groupSelection })
+        );
+      }
+
+      onGroupChange?.({
+        tableId: groupingId,
+        groupByField: groupSelection,
+        groupByFields: newSelectedGroups,
```

Comment:
enhanced `onGroupChange` to add `groupByFields`, so the consumer can know all the current selected groups as part of the `onChange` method, and that adds flexibility for reusing it for other purposes (such as setting the current state on the URL as done in this PR on the cloud security code).

Reply:
onGroupChange could be x-pack/plugins/security_solution/public/detections/components/alerts_table/alerts_grouping.tsx line 79, but its type doesn't include the `groupByFields` (definded in x-pack/plugins/security_solution/public/common/lib/telemetry/events/alerts_grouping/types.ts line 11) so we see [this error](https://github.com/elastic/kibana/pull/175749#issuecomment-1916654182)

This should be fixed by updating x-pack/plugins/security_solution/public/detections/components/alerts_table/alerts_grouping.tsx line 80 to take only groupByField and tableId

Reply:
Hi @angorayc, thanks for catching this, I wasn't aware there was also a telemetry in the `alerts_grouping.tsx` relying on the onChange, I fixed it as you suggested and the schema validation error is gone, I also added a unit test for this part of the code